examples/structured_data/classification_with_grn_and_vsn.py

```diff
@@ -0,0 +1,413 @@
+"""
+Title: Classification with Gated Residual and Variable Selection Networks
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/15
+Last modified: 2021/01/15
+Description: Using Gated Residual and Variable Selection Networks for income level prediction.
+"""
+
+"""
+## Introduction
+
+This example demonstrates the use of Gated
+Residual Network (GRN) and Variable Selection Networks (VSN), proposed by
+[Bryan Lim et al.](https://arxiv.org/abs/1912.09363), for structured data
```

Comment:
Suggestions:

- Add the name of the paper ("Temporal Fusion Transformers for Interpretable Multi-horizon Time Series Forecasting by Bryan Lim et al..."), since it's not named after Gated Residual Networks (GRNs) and Variable Selection Networks (VSNs) your example is based on.
  - If your example is only about GRNs and not the TFT model (the main thesis in the paper you're referencing), this should be made clearer, I think. 
   
GRNs inside the TFT model:
<img width="833" alt="image" src="https://user-images.githubusercontent.com/19637339/103161478-3bffe200-47da-11eb-9f26-ce141414c0d4.png">

- The TFT paper is from 2019 but Gated ResNets (GRNs) date back to at least 2018 (https://web.cse.ohio-state.edu/~wang.77/papers/TCW.icassp18.pdf) - can you check again that GRN/VSN were proposed in 2019 in the TFT paper?
- Variable selection with neural nets were apparently proposed back in 1996: https://www.sciencedirect.com/science/article/abs/pii/0925231295001212 - can you double check your source again?

Reply:
I see that the TFT paper's authors claim to propose a GRN but they may not be the first ones, unless their version only shares the name.

The 2018 paper submitted for IEEE (https://web.cse.ohio-state.edu/~wang.77/papers/TCW.icassp18.pdf) even says: _"In this study, we have proposed a gated residual network with dilated convolutions, named GRN..."_

🤷

Reply:
@8bitmp3 - Thank you for your comments. I made it clear in the introduction that the example covers only GRNs an VSNs, rather than the whole TFT model described in the paper. I also added the paper title (the reason I wanted to avoid adding the paper name is that it can be confusing from the first sight as the paper is about time-series forecasting, unlike the example). I still prefer to put the author with a like to the paper, rather than the title.

Also, I checked the same references you shared with me as I was preparing the example :-) The TFT paper uses the same names but different architectures and applications of these models.